src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs

```diff
@@ -148,10 +148,10 @@ private string NormalizeFilter(string filter)
 
         private bool IsDirectoryPath(string path)
         {
-            return path != null && path.Length > 1 &&
-                (path[path.Length - 1] == Path.DirectorySeparatorChar ||
-                path[path.Length - 1] == Path.AltDirectorySeparatorChar);
-        }
+			if (path == null)
+				throw new ArgumentNullException();
```

Comment:
@troydai thanks, I will remove the ArgumentNullException and replace it with null check